### PR TITLE
homework week 3- Tho Quyen Ngo

### PR DESCRIPTION
**Pull request by willngo36**

### CODE REVIEW

File: homeworks/week3/Quyen-Ngo.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+static class SinglyLinkedList {
+	private Node head = null;
+    
+    // Add to a specific position; do nothing if the index is out of range or e is null
+	//Add function: Time Complexity O(n)
+	//Add function: Space Complexity O(1)
+    public void add(int index, Node e) {
+
```
Comment (ubriela) [inline homeworks/week3/Quyen-Ngo.java:32]: Missing the checks for out of range cases: index = -1 or index > len.
We should do nothing in those cases.

File: homeworks/week3/Quyen-Ngo.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    	e.next = curr;
+    	prev.next = e;
+    }
+
+    // Removes the first occurrence of the specified element from this list if it is present (i.e., its value equals to e.value.
+    // Returns true if this list contained the specified element; otherwise, returns false.
+    
+    //Remove function: Time Complexity O(n)
+    //Remove function: Space Complexity O(1)
+    public boolean remove(Node e) {
```
Comment (ubriela) [inline homeworks/week3/Quyen-Ngo.java:56]: Check edge case where e = null.

**Review by ubriela — COMMENTED** [review]
Nice solutions overall. Adding some more test cases would allow you to catch those edge cases.
